### DISCUSSION TITLE
Error importing CAD Expressions file

### DISCUSSION THREAD

**Discussion by JimSulsona** [Cookie-AutoDelete/Cookie-AutoDelete#1461] [Q&A / Support]
I exported a file from one of my computers so that I could import it into a new computer, and it gives me this message:
Error: File given failed validation. Missing identifier 'settings': CAD_Expressions_2022-11-14_09.23.05.json
Don't have any idea what to do...

**Comment by kennethtran93** [comment]
What you've exported is just the list of expressions, not the core settings.  You can import that file from the list of expressions page.

There is a separate export just for the core settings.

**Comment by JimSulsona** [comment]
OK, I am clearly confused.  What I did was go to my original computer's CAD Settings page and select "Export Core Settings".  Then I went to the new computer's identical page and selected "Import Core Settings".  That's when I get the error message.  What am I doing wrong?

By the way, I realized that my new Firefox browser is Synced to my Firefox account - does that make any difference?  I thought it would provide all the same settings, but it doesn't know any of the saved website CAD settings that I have on my original machine.  It did give me all my bookmarks, etc.  Just not sure what else I need to do.  Your help is greatly appreciated!

**Comment by kennethtran93** [comment]
We have not implemented any data sync yet, so this doesn't affect us.

After reading the process you provided, I think you may have used the wrong import button/page.  Your Expression file needs to be imported from the List of Expressions page, not the Settings page. 
![image](https://user-images.githubusercontent.com/6724477/202384371-cbbb4999-a6b6-417e-8a35-ae174aaac971.png)



If not...here's some more details.  When you export the core settings, the filename provided would have included `CoreSettings` as shown:

![image](https://user-images.githubusercontent.com/6724477/202382137-987a2b66-37b3-4d0a-a4d7-610cddebc975.png)

On the save window popup prompt, you may have accidentally clicked on the other json file meant for expressions instead of simply pressing Enter or Save when prompted, which in turn overwrote the contents inside that file.  If the filename value matches an existing file, it will overwrite the file (well it would prompt you to confirm you wish to overwrite that file first).  If it doesn't match the file, it will create a new one.

For comparison, here's that a sample of the contents within each file should be like after exporting them, left side being the `CoreSettings` and the right side being `Expressions` as shown by the filename tabs:
![image](https://user-images.githubusercontent.com/6724477/202383531-278c5624-9cef-49a9-a08c-7c9971c4232c.png)

**Comment by JimSulsona** [comment]
It's entirely possible that I may have mixed up the two different export options!  I'll have to go back to the original computer and retry the export process to see if I have the same results.

One question...if I successfully import the files, does it completely replace my existing settings, or does it append to my existing settings?

**Reply by kennethtran93** [reply]
For importing expressions, if it finds an exact expression match, it will ignore it, otherwise will import it as a new expression.

For importing Core Settings, it will update the setting if it is a different value.  If you only wish to update specific values while keeping the rest as is, you'll need to manually go into the json file and delete the entries you don't want updated.
